src/main/java/net/fabricmc/tinyremapper/mixin/annotation/MixinAnnotationVisitor.java

```diff
@@ -0,0 +1,44 @@
+package net.fabricmc.tinyremapper.mixin.annotation;
+
+import java.util.Objects;
+
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.commons.Remapper;
+
+import net.fabricmc.tinyremapper.TinyRemapper;
+
+class MixinAnnotationVisitor extends SkippableAnnotationVisitor implements MixinAnnotationVisitorLogger {
```

Comment:
Hmm, I think you might store the key to handler to a map (somewhat like mojang dfu?!), and you may generalize different strings to different data types (such as class names, descriptors, etc)

Reply:
I suspect LogicFan will switch to ClassNode sooner than later ;)